tests/python_package_test/test_basic.py

```diff
@@ -89,6 +91,115 @@ def test_basic(tmp_path):
                                    bst.predict, tname)
 
 
+class NumpySequence(lgb.Sequence):
+    def __init__(self, ndarray, batch_size):
+        self.ndarray = ndarray
+        self.batch_size = batch_size
+
+    def __getitem__(self, idx):
+        # The simple implementation is just a single "return self.ndarray[idx]"
+        # The following is for demo and testing purpose.
+        if isinstance(idx, numbers.Integral):
+            return self._get_one_line(idx)
+        elif isinstance(idx, slice):
+            if not (idx.step is None or idx.step is 1):
+                raise NotImplementedError("No need to implement, caller will not set step by now")
+            return self.ndarray[idx.start: idx.stop]
+        else:
+            raise TypeError(f"Sequence Index must be an integer/list/slice, got {type(idx)}")
+
+    def _get_one_line(self, idx):
+        return self.ndarray[idx]
```

Comment:
The purpose here is to serve as an example and make it more clear what the implementation should do.

I'm going to remove this because this should be very easy to understand even without this function.